src/main/java/org/mozilla/magnet/net/scanner/mdns/ScannerMDNS.java

```diff
@@ -49,22 +50,18 @@ public void onDiscoveryStarted(String regType) {
 
             @Override
             public void onServiceFound(NsdServiceInfo service) {
-                // A service was found!  Do something with it
-                Log.d(TAG, "Service discovery success" + service);
+                Log.d(TAG, "service found: " + service);
                 String name = service.getServiceName();
-                if (URLUtil.isNetworkUrl(name)) {
-                    JSONObject result = new JSONObject();
-                    try {
-                        result.put("url", name);
-                        JSONObject metadata = new JSONObject();
-                        metadata.put("type", service.getServiceType());
-                        result.put("metadata", metadata);
-
-                        MDNSScanner.this.notify(result);
-                    } catch (JSONException e) {
-                        Log.e(TAG, "Could not create response object for " + name);
-                    }
+
+                // not all mdns advertisements are urls
+                if (!URLUtil.isNetworkUrl(name)) {
+                    return;
                 }
+
+                MagnetScannerItem item = new MagnetScannerItem();
+                item.setUrl(name);
+                item.setType(service.getServiceType());
```

Comment:
Is this necessary?

Is not going to be overwritten here: https://github.com/mozilla-magnet/magnet-scanner-android/pull/5/files#diff-8a3b8412dec49dc1c0fe3b0ed1d7bdb7R45